### ISSUE TITLE
Conflict on build link creation should fail with a HTTP 400 error

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#940]
Root error:

```
INSERT INTO BUILD_LINKS(BUILDID, TARGETBUILDID) VALUES (?, ?)]; ERROR: duplicate key value violates unique constraint "build_links_uq"
```

Root cause: the same link is tried to be created twice.

This should be a user exception, not an internal error.

Additionally, an atomic insert/update is needed here.

**Comment by dcoraboeuf** [comment]
Available in 3.43.24

**Comment by dcoraboeuf** [comment]
Available in 4.1.9
